tests/test_adapter.py

```diff
@@ -90,7 +94,8 @@ def setUp(self) -> None:
                                                            b'permitted_nationalities=IRL)&doctor:and&admin:and&(Apple:or|'
                                                            b'SOMETHING:or))'),
                                         ('Exec-Path', b'Automatic Adapter-to-In-Memory List'),
-                                        ('Request-Id', b'List:uuid4'), ('traceparent', b'')
+                                        ('Request-Id', b'List:uuid4'), ('traceparent', b''),
+                                        ('Data-Source-Name', b'None'), ('Data-Source-Type', b'None')
                                         ]
 
```

Comment:
A test here to ensure the headers are not included with None values would be good.